corpus_studies/solidity_state_study/slither_plugins/run_tests.py

```diff
@@ -1,41 +1,27 @@
-from hasstate import ContractStateDetector
+from ContractStateDetector import ContractStateDetector
+from ..tests.set_solc_version import set_version
 import glob
+import pytest
 from slither import Slither
-from slither.detectors.state.hasstate import HasState
 
-# For those in the state directory, we expect at least one contract to be detected with state.
-# For those in the nostate directory, we expect none of the contracts to b  e detected with state.
-def test_state():
-    failed = []
-    files = [f for f in glob.glob("tests/state/*.sol")]
-    for f in files:
-        print("testing %s..." % f)
-        slither = Slither(f)
-        if not any(ContractStateDetector(c).is_stateful_contract() for c in slither.contracts):
-            failed.append(f)
-    return failed
+# To run tests, run python3 -m pytest run_tests.py
```

Comment:
I'm a python newbie — can one also invoke pytest directly (https://docs.pytest.org/en/6.2.x/)?

Reply:
Yes, I just fixed the imports so you can do that.